chain/ethereum/src/codec.rs

```diff
@@ -295,3 +299,37 @@ impl Into<EthereumBlockWithCalls> for &Block {
         }
     }
 }
+
+impl From<Block> for BlockPtr {
+    fn from(b: Block) -> BlockPtr {
+        (&b).into()
+    }
+}
+
+impl<'a> From<&'a Block> for BlockPtr {
+    fn from(b: &'a Block) -> BlockPtr {
+        BlockPtr::from((H256::from_slice(b.hash.as_ref()), b.number))
+    }
+}
+
+impl BlockchainBlock for Block {
+    fn number(&self) -> i32 {
+        BlockNumber::try_from(self.number).unwrap()
+    }
+
+    fn ptr(&self) -> BlockPtr {
+        self.into()
+    }
+
+    fn parent_ptr(&self) -> Option<BlockPtr> {
+        let parent_hash = &self.header.as_ref().unwrap().parent_hash;
+
+        match parent_hash.len() {
+            0 => None,
```

Comment:
Actually the field is probably not populated at all in the protobuf serialization format at all, but since a `parent_hash` is backed in Rust by `::prost::alloc::vec::Vec<u8>`, `Vec` here is the empty vector.